### PR TITLE
[SPARK-32959][SQL][TEST] Fix an invalid test in DataSourceV2SQLSuite

### PR DESCRIPTION
**Pull request by imback82**
### What changes were proposed in this pull request?

This PR addresses two issues related to the `Relation: view text` test in `DataSourceV2SQLSuite`.

1. The test has the following block:
```scala
withView("view1") { v1: String =>
  sql(...)
}
```
Since `withView`'s signature is `withView(v: String*)(f: => Unit): Unit`, the `f` that will be executed is ` v1: String => sql(..)`, which is just defining the anonymous function, and _not_ executing it.

2. Once the test is fixed to run, it actually fails. The reason is that the v2 session catalog implementation used in tests does not correctly handle `V1Table` for views in `loadTable`. And this results in views resolved to `ResolvedTable` instead of `ResolvedView`, causing the test failure: https://github.com/apache/spark/blob/f1dc479d39a6f05df7155008d8ec26dff42bb06c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L1007-L1011

### Why are the changes needed?

Fixing a bug in test.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing test.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala [hunk 1]
```diff
@@ -47,10 +47,7 @@ private[connector] trait TestV2SessionCatalogBase[T <: Table] extends Delegating
       tables.get(ident)
     } else {
       // Table was created through the built-in catalog
-      val t = super.loadTable(ident)
-      val table = newTable(t.name(), t.schema(), t.partitioning(), t.properties())
-      tables.put(ident, table)
-      table
```
Comment (imback82) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala:53]: Now that the test actually runs, it actually fails because this block remove `V1Table` info, which is needed in the following:
https://github.com/apache/spark/blob/f1dc479d39a6f05df7155008d8ec26dff42bb06c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L1008

I don't think we need to put the result of `loadTable` into `tables`, but if we need to cache it, I can change this to something like:
```scala
super.loadTable(ident) match {
  case v1Table: V1Table if v1Table.v1Table.tableType == CatalogTableType.VIEW => v1Table // or 'case v1Table: V1Table => v1Table'
  case t =>
    val table = newTable(t.name(), t.schema(), t.partitioning(), t.properties())
    tables.put(ident, table)
    table
```

File: sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala [hunk 2]
```diff
@@ -47,10 +47,13 @@ private[connector] trait TestV2SessionCatalogBase[T <: Table] extends Delegating
...
+      super.loadTable(ident) match {
+        case v1Table: V1Table if v1Table.v1Table.tableType == CatalogTableType.VIEW => v1Table
+        case t =>
+          val table = newTable(t.name(), t.schema(), t.partitioning(), t.properties())
+          tables.put(ident, table)
+          table
+      }
```
Comment (imback82) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala:56]: Now that the test actually runs, it actually fails because this block remove `V1Table` info, which is needed in the following:
https://github.com/apache/spark/blob/f1dc479d39a6f05df7155008d8ec26dff42bb06c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L1008

**Comment by dongjoon-hyun** [comment]
In this case, could you use JIRA id please, @imback82 ?

**Comment by imback82** [comment]
@dongjoon-hyun, I updated the PR title with JIRA id. Does the fixed test need to have it as well - e.g., `test("SPARK-32959: Relation: view text")`?

**Comment by imback82** [comment]
cc @cloud-fan

File: sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala [hunk 3]
```diff
@@ -47,10 +47,13 @@ private[connector] trait TestV2SessionCatalogBase[T <: Table] extends Delegating
       tables.get(ident)
     } else {
       // Table was created through the built-in catalog
-      val t = super.loadTable(ident)
-      val table = newTable(t.name(), t.schema(), t.partitioning(), t.properties())
-      tables.put(ident, table)
-      table
+      super.loadTable(ident) match {
+        case v1Table: V1Table if v1Table.v1Table.tableType == CatalogTableType.VIEW => v1Table
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala:51]: @imback82 . This is orthogonal to `Fix an invalid test in DataSourceV2SQLSuite`. Could you make a PR for this one independently?

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala [hunk 4]
```diff
@@ -758,8 +758,9 @@ class DataSourceV2SQLSuite
 
   test("Relation: view text") {
     val t1 = "testcat.ns1.ns2.tbl"
+    val v1 = "view1"
     withTable(t1) {
-      withView("view1") { v1: String =>
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:762]: what was the value of `v1` before?

File: sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala (hunk 3 above)
Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala:51]: or we should at least mention it in the PR description, to explain why it's needed to fix the test.

Comment (imback82) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/TestV2SessionCatalogBase.scala:51]: Updated the PR description.

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala (hunk 4 above)
Comment (imback82) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala:762]: This just defines an anonymous function, and `v1` is a function parameter.

**Comment by cloud-fan** [comment]
good catch! merging to master!
